python/lsst/pipe/tasks/imageDifference.py

```diff
@@ -970,15 +970,26 @@ def run(self, exposure=None, selectSources=None, templateExposure=None, template
                 # useScoreImageDetection=True and doDecorrelation=True is not allowed in the config
                 if self.config.doDecorrelation and self.config.doSubtract:
                     preConvKernel = None
-                    if preConvPsf is not None:
+                    if self.config.useGaussianForPreConvolution:
                         preConvKernel = preConvPsf.getLocalKernel()
-                    decorrResult = self.decorrelate.run(exposureOrig, subtractRes.warpedExposure,
-                                                        subtractedExposure,
-                                                        subtractRes.psfMatchingKernel,
-                                                        spatiallyVarying=self.config.doSpatiallyVarying,
-                                                        preConvKernel=preConvKernel,
-                                                        templateMatched=self.config.convolveTemplate)
-                    subtractedExposure = decorrResult.correctedExposure
+                    if self.config.useScoreImageDetection:
+                        decorrResult = self.decorrelate.run(exposureOrig, subtractRes.warpedExposure,
+                                                            scoreExposure,
+                                                            subtractRes.psfMatchingKernel,
+                                                            spatiallyVarying=self.config.doSpatiallyVarying,
+                                                            preConvKernel=preConvKernel,
+                                                            templateMatched=True,
+                                                            preConvMode=True)
+                        scoreExposure = decorrResult.correctedExposure
+                    else:
+                        decorrResult = self.decorrelate.run(exposureOrig, subtractRes.warpedExposure,
+                                                            subtractedExposure,
+                                                            subtractRes.psfMatchingKernel,
+                                                            spatiallyVarying=self.config.doSpatiallyVarying,
+                                                            preConvKernel=None,
+                                                            templateMatched=self.config.convolveTemplate,
+                                                            preConvMode=False)
+                        subtractedExposure = decorrResult.correctedExposure
```

Comment:
I think we want to decorrelate the subtracted exposure, even if we use the score image for detection.